handler.go

```diff
@@ -17,16 +17,16 @@ type Redriver struct {
 	Debug            bool
 }
 
-type processResult struct {
-	message events.SQSMessage
-	err     error
-}
-
 // MessageProcessor is the required function signature for processors.
 type MessageProcessor = func(event events.SQSMessage) error
 
-func (redriver Redriver) deleteProcessedMessages(processedMessages *[]processResult, sqsConnector *sqs.SQS) error {
-	for _, processedMessage := range *processedMessages {
+func (redriver Redriver) deleteProcessedMessages(processedMessages processResults, sqsConnector *sqs.SQS) error {
+	for _, processedMessage := range processedMessages.successResults {
+		if redriver.Debug {
+			log.Debugf("Debug mode enabled. Mocking deletion of message with ID %s", processedMessage.message.MessageId)
+			continue
+		}
```

Comment:
@qraimbault i had to move the Debug check as it was throwing a panic error instead of a standard error. I added some log to be sure that the deleteProcessedMessages is still being called even when we are in Debug mode.